pagefaultgames/pokerogue
# [Ability] Implement the secondary effect with Super Luck and Compound Eyes

> [!Warning]
> The PR is now maintained by @DayKev 





## What are the changes?

Added the Gen VIII out of battle effect for Super Luck, and Compound Eyes:
### If a Pokémon with this Ability leads the [party](https://bulbapedia.bulbagarden.net/wiki/Party) (even if [fainted](https://bulbapedia.bulbagarden.net/wiki/Fainting)), the chances of finding [a wild Pokémon holding an item](https://bulbapedia.bulbagarden.net/wiki/List_of_Pok%C3%A9mon_by_wild_held_item) increase from 50%/5% to 60%/20%. This effect is shared with [Compound Eyes](https://bulbapedia.bulbagarden.net/wiki/Compound_Eyes_(Ability)).

## Why am I doing these changes?



Both Super Luck and Compound Eyes are partially implemented, so I thought I would fix them.

## What did change?


I added the 20% extra enemy held item modifier, so if you give `BonusItemChance` to an ability, it should make enemies 20% more likely to have a held item.

### Screenshots/Videos


Video of the chances of an enemy holding an item with Super Luck/Compound Eyes.

[Video](https://github.com/pagefaultgames/pokerogue/assets/71776311/e7d75b24-b05b-4781-a597-60e86e594346)

Video of the chances of an enemy holding an item without Super Luck/Compound Eyes.

[Video](https://github.com/pagefaultgames/pokerogue/assets/71776311/eb291c48-26cd-4f2a-bf1c-c61de8cad41c)

## How to test the changes?




Give yourself the ability Super Luck/Compound Eyes, and set some console.logs before the chances are set/changed in battle-phase.ts, and after they are set/changed. Check `BonusItemChance` documentation for more information.

## Checklist
